platform/openide.util.ui/src/org/openide/util/Utilities.java

```diff
@@ -1357,17 +1357,50 @@ private static Frame findMainWindow()
      */
     // PR4739
     public static Component findDialogParent() {
-        Component parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+        return findDialogParent(null);
+    }
+
+    /**
+     * Finds an appropriate component to use for a dialog's parent. Similar to {@link #findDialogParent()}
+     * with the ability to specify a suggested parent component.
+     *
+     * @param suggestedParent the component to return if suitable
+     * @return A suitable parent component for swing dialog displayers.
+     *
+     * @see #findMainWindow()
+     * @since 9.30
+     */
+    public static Component findDialogParent(Component suggestedParent) {
```

Comment:
Don't see `Component findDialogParent(Component suggestedParent)` used anywhere in the PR. You're suggesting adding it to the API? I see how it's convenient to check if the suggestedParent is unsuitable because there's a modal dialog active.

Reply:
> You're suggesting adding it to the API?

Yes, would be useful when using a `JOptionPane` (or similar) where you want the new dialog to be positioned relative to a specific component (`suggestedParent`) rather than just the containing window. If `NotifyDescriptor` had the option to specify a suggested parent, `DialogDisplayerImpl` would use this overload.